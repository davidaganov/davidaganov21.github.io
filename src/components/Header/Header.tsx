import { useTranslation } from "react-i18next"
import { HeaderProps } from "./Header.props"
import { Button } from "../"
import styles from "./Header.module.sass"
import cn from "classnames"

export const Header = ({ ...props }: HeaderProps) => {
  const { t } = useTranslation()

  return (
    <header
      className={styles.header}
      {...props}
    >
      <div className={cn(styles.inner, "inner")}>
        <div className={styles.content}>
          <div className={cn(styles.blob, styles.first)}></div>
          <div className={cn(styles.blob, styles.second)}></div>
          <div className={cn(styles.blob, styles.third)}></div>

          <p className={styles.subtitle}>{t("header.subtitle")}</p>
          <h1 className={styles.title}>{t("header.name")}</h1>
          <p className={styles.description}>{t("header.description")}</p>
          <Button
            href="#footer"
            className={styles.btn}
          >
            {t("header.contact")}
          </Button>
        </div>
      </div>
    </header>
  )
}
