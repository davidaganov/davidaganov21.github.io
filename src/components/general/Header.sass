.header
  position: relative
  height: 100vh
  max-height: 108rem
  min-height: 60rem
  &__inner
    position: relative
    display: flex
    flex-direction: column
    height: 100%

  &__content
    margin: auto 0

  &__title
    margin-top: 1rem
    color: var(--color-white-700)
    strong
      display: block
      color: var(--color-white)

  &__subtitle
    color: var(--color-accent)
    font-family: var(--second-font)

  &__description
    position: relative
    margin-top: 2rem
    color: var(--color-white-700)
    @media (min-width: $medium-min)
      max-width: 60rem
      padding-left: 2rem
    @media (max-width: $medium-max)
      padding-left: 1rem
    &::before
      position: absolute
      content: ""
      width: .1rem
      height: 100%
      top: 0
      left: 0
      background-color: var(--color-accent)

  &__link-icon
    width: 18px
    height: 18px
    margin: 0 5px -2px 0
    opacity: .8

  &__btn
    margin-top: 4.5rem
