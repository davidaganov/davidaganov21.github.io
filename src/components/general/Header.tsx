import Button from "../base/Button"
import Navbar from "./Navbar"

export default function Header() {
  return (
    <header className="header">
      <div className="header__inner inner">
        <Navbar />

        <div className="header__content">
          <p className="header__subtitle">Hello, my name is</p>
          <h1 className="header__title">
            <strong>David Aganov</strong>I build things for the web
          </h1>
          <p className="header__description">
            I’m middle frontend developer. I like to educate new technologies and try new things in
            every possible way. At the moment I’m working in mobile games studio{" "}
            <a
              className="header__link inline-link"
              href="https://www.linkedin.com/company/sabgames/mycompany/"
            >
              <svg
                className="header__link-icon"
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 24 24"
                fill="currentColor"
              >
                <path d="M19 0H5a5 5 0 0 0-5 5v14a5 5 0 0 0 5 5h14a5 5 0 0 0 5-5V5a5 5 0 0 0-5-5zM8 19H5V8h3v11zM6.5 6.73a1.76 1.76 0 0 1 0-3.53c.97 0 1.75.8 1.75 1.77S7.47 6.73 6.5 6.73zM20 19h-3v-5.6c0-3.37-4-3.12-4 0V19h-3V8h3v1.76a3.8 3.8 0 0 1 7 2.48V19z" />
              </svg>
              SAB Games
            </a>
          </p>
          <Button
            classes="header__btn"
            value="Button"
          />
        </div>
      </div>
    </header>
  )
}
