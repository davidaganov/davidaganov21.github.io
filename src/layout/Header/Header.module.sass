.header
  position: relative
  height: 100vh
  max-height: 108rem
  min-height: 60rem
  overflow: hidden

.inner
  position: relative
  display: flex
  flex-direction: column
  height: 100%

.content
  position: relative
  margin: auto 0

.title
  margin-top: 1.5rem
  color: var(--color-white-700)
  letter-spacing: -.03em
  strong
    display: block
    color: var(--color-white)

.subtitle
  color: var(--color-accent)
  font: 400 1.1rem/1.9rem var(--second-font)
  text-transform: uppercase

.description
  position: relative
  margin-top: 2rem
  color: var(--color-white-700)
  @media (min-width: 769px)
    max-width: 60rem
    padding-left: 2rem
  @media (max-width: 768px)
    max-width: 50rem
    padding-left: 1rem
  &::before
    position: absolute
    content: ""
    width: .2rem
    height: 100%
    top: 0
    left: 0
    background-color: var(--color-accent)

.link-icon
  width: 1.8rem
  height: 1.8rem
  margin: 0 .5rem -.2rem 0
  opacity: .8

.btn
  margin-top: 6rem

.blob
  position: absolute
  background-color: var(--color-accent)
  border-radius: 30% 50% 20% 40%
  opacity: .15
  z-index: -1

.first
  height: 20rem
  width: 20rem
  left: 70%
  top: 80%
  animation: transform 20s ease-in-out infinite both alternate, movement_one 40s ease-in-out infinite both

.second
  top: -5rem
  transform: rotate(-180deg)
  animation: transform 30s ease-in-out infinite both alternate, movement_two 60s ease-in-out infinite both
  @media (min-width: 1000px)
    height: 50rem
    width: 50rem
    left: -15rem
  @media (max-width: 999px) and (min-width: 576px)
    width: 40rem
    height: 40rem
    left: -5rem
  @media (max-width: 575px)
    width: 35rem
    height: 35rem
    left: -10rem

.third
  @media (min-width: 769px)
    height: 45rem
    width: 45rem
    left: 50rem
    top: -15rem
    transform: rotate(-180deg)
    animation: transform 30s ease-in-out infinite both alternate, movement_two 60s ease-in-out infinite both
  @media (max-width: 768px)
    display: none

@keyframes transform
  0%,
  100%
    border-radius: 33% 67% 70% 30% / 30% 30% 70% 70%
  20%
    border-radius: 37% 63% 51% 49% / 37% 65% 35% 63%
  40%
    border-radius: 36% 64% 64% 36% / 64% 48% 52% 36%
  60%
    border-radius: 37% 63% 51% 49% / 30% 30% 70% 70%
  80%
    border-radius: 40% 60% 42% 58% / 41% 51% 49% 59%

@keyframes movement_one
  0%,
  100%
    transform: none
  50%
    transform: translate(-50%, 20%) rotateY(10deg) scale(1.2)

@keyframes movement_two
  0%,
  500%
    transform: none
  50%
    transform: translate(30%, 10%) rotate(-200deg) scale(1.2)
