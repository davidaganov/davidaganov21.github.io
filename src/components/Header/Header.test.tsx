import { render, screen } from "@testing-library/react"
import { Header } from "./Header"

const data = [
  { link: "#test", title: "Test" },
  { link: "#link", title: "Link" }
]

describe("Header component", () => {
  it("Header renders", () => {
    render(<Header list={data} />)

    expect(screen.getByText(/david aganov/i)).toBeInTheDocument()
  })

  it("Header snapshot", () => {
    const title = render(<Header list={data} />)

    expect(title).toMatchSnapshot()
  })
})
