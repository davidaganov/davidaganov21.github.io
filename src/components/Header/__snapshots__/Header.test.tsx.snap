// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="inner inner"
        >
          <header
            class="header"
          >
            <div
              class="inner inner"
            >
              <button
                aria-expanded="false"
                aria-label="Open navbar"
                class="burger"
                tabindex="0"
                type="button"
              >
                <svg>
                  burger.svg
                </svg>
              </button>
              <nav
                class="navbar"
                role="navigation"
              >
                <button
                  aria-label="Close navbar"
                  class="close"
                  tabindex="-1"
                  type="button"
                >
                  <svg>
                    close.svg
                  </svg>
                </button>
                <ul
                  class="menu"
                >
                  <li
                    class="item"
                  >
                    <span>
                      0
                      1
                      .
                    </span>
                    <a
                      class="link inline-link inline-link--white"
                      href="#test"
                    >
                      Test
                    </a>
                  </li>
                  <li
                    class="item"
                  >
                    <span>
                      0
                      2
                      .
                    </span>
                    <a
                      class="link inline-link inline-link--white"
                      href="#link"
                    >
                      Link
                    </a>
                  </li>
                </ul>
                <div
                  class="switcher"
                  role="listbox"
                >
                  <span
                    class="visually-hidden"
                    id="selectLang"
                  >
                    Select language
                  </span>
                  <button
                    aria-label="English"
                    aria-labelledby="selectLang lang-en"
                    aria-selected="true"
                    class="btn btnActive"
                    id="lang-en"
                    role="option"
                    type="button"
                  >
                    en
                  </button>
                  <button
                    aria-label="Russian"
                    aria-labelledby="selectLang lang-ru"
                    aria-selected="false"
                    class="btn"
                    id="lang-ru"
                    role="option"
                    type="button"
                  >
                    ru
                  </button>
                </div>
              </nav>
              <button
                aria-hidden="true"
                class="backdrop"
                tabindex="-1"
                type="button"
              />
            </div>
          </header>
          <div
            class="content"
          >
            <div
              class="blob first"
            />
            <div
              class="blob second"
            />
            <div
              class="blob third"
            />
            <p
              class="subtitle"
            >
              Hello, my name is
            </p>
            <h1
              class="title"
            >
              <strong>
                David Aganov
              </strong>
              I build things for the web
            </h1>
            <p
              class="description"
            >
              I’m frontend developer. I like to learn new technologies and try new things in every possible way. At the moment I’m working in mobile games studio 
              <a
                aria-label="The company I work for is SAB Games"
                class="inline-link"
                href="https://www.linkedin.com/company/sabgames/mycompany/"
              >
                <svg
                  class="link-icon"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 0H5a5 5 0 0 0-5 5v14a5 5 0 0 0 5 5h14a5 5 0 0 0 5-5V5a5 5 0 0 0-5-5zM8 19H5V8h3v11zM6.5 6.73a1.76 1.76 0 0 1 0-3.53c.97 0 1.75.8 1.75 1.77S7.47 6.73 6.5 6.73zM20 19h-3v-5.6c0-3.37-4-3.12-4 0V19h-3V8h3v1.76a3.8 3.8 0 0 1 7 2.48V19z"
                  />
                </svg>
                SAB Games
              </a>
            </p>
            <a
              class="btn btn"
              href="#footer"
            >
              Contact me
            </a>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="inner inner"
      >
        <header
          class="header"
        >
          <div
            class="inner inner"
          >
            <button
              aria-expanded="false"
              aria-label="Open navbar"
              class="burger"
              tabindex="0"
              type="button"
            >
              <svg>
                burger.svg
              </svg>
            </button>
            <nav
              class="navbar"
              role="navigation"
            >
              <button
                aria-label="Close navbar"
                class="close"
                tabindex="-1"
                type="button"
              >
                <svg>
                  close.svg
                </svg>
              </button>
              <ul
                class="menu"
              >
                <li
                  class="item"
                >
                  <span>
                    0
                    1
                    .
                  </span>
                  <a
                    class="link inline-link inline-link--white"
                    href="#test"
                  >
                    Test
                  </a>
                </li>
                <li
                  class="item"
                >
                  <span>
                    0
                    2
                    .
                  </span>
                  <a
                    class="link inline-link inline-link--white"
                    href="#link"
                  >
                    Link
                  </a>
                </li>
              </ul>
              <div
                class="switcher"
                role="listbox"
              >
                <span
                  class="visually-hidden"
                  id="selectLang"
                >
                  Select language
                </span>
                <button
                  aria-label="English"
                  aria-labelledby="selectLang lang-en"
                  aria-selected="true"
                  class="btn btnActive"
                  id="lang-en"
                  role="option"
                  type="button"
                >
                  en
                </button>
                <button
                  aria-label="Russian"
                  aria-labelledby="selectLang lang-ru"
                  aria-selected="false"
                  class="btn"
                  id="lang-ru"
                  role="option"
                  type="button"
                >
                  ru
                </button>
              </div>
            </nav>
            <button
              aria-hidden="true"
              class="backdrop"
              tabindex="-1"
              type="button"
            />
          </div>
        </header>
        <div
          class="content"
        >
          <div
            class="blob first"
          />
          <div
            class="blob second"
          />
          <div
            class="blob third"
          />
          <p
            class="subtitle"
          >
            Hello, my name is
          </p>
          <h1
            class="title"
          >
            <strong>
              David Aganov
            </strong>
            I build things for the web
          </h1>
          <p
            class="description"
          >
            I’m frontend developer. I like to learn new technologies and try new things in every possible way. At the moment I’m working in mobile games studio 
            <a
              aria-label="The company I work for is SAB Games"
              class="inline-link"
              href="https://www.linkedin.com/company/sabgames/mycompany/"
            >
              <svg
                class="link-icon"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 0H5a5 5 0 0 0-5 5v14a5 5 0 0 0 5 5h14a5 5 0 0 0 5-5V5a5 5 0 0 0-5-5zM8 19H5V8h3v11zM6.5 6.73a1.76 1.76 0 0 1 0-3.53c.97 0 1.75.8 1.75 1.77S7.47 6.73 6.5 6.73zM20 19h-3v-5.6c0-3.37-4-3.12-4 0V19h-3V8h3v1.76a3.8 3.8 0 0 1 7 2.48V19z"
                />
              </svg>
              SAB Games
            </a>
          </p>
          <a
            class="btn btn"
            href="#footer"
          >
            Contact me
          </a>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
